service/Core/DataFormats/Office/MsExcelDecoder.cs

```diff
@@ -1,70 +1,54 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
 using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 using ClosedXML.Excel;
+using Microsoft.Extensions.Logging;
+using Microsoft.KernelMemory.Diagnostics;
+using Microsoft.KernelMemory.Pipeline;
 
 namespace Microsoft.KernelMemory.DataFormats.Office;
 
-public class MsExcelDecoder
+public class MsExcelDecoder : IContentDecoder
 {
-    private const string DefaultSheetNumberTemplate = "\n# Worksheet {number}\n";
-    private const string DefaultEndOfSheetTemplate = "\n# End of worksheet {number}";
-    private const string DefaultRowPrefix = "";
-    private const string DefaultColumnSeparator = ", ";
-    private const string DefaultRowSuffix = "";
-    private const string DefaultBlankCellValue = "";
-
-    private readonly bool _withWorksheetNumber;
-    private readonly bool _withEndOfWorksheetMarker;
-    private readonly bool _withQuotes;
-    private readonly string _worksheetNumberTemplate;
-    private readonly string _endOfWorksheetMarkerTemplate;
-    private readonly string _rowPrefix;
-    private readonly string _columnSeparator;
-    private readonly string _rowSuffix;
-    private readonly string _blankCellValue;
-
-    public MsExcelDecoder(
-        bool withWorksheetNumber = true,
-        bool withEndOfWorksheetMarker = false,
-        bool withQuotes = true,
-        string? worksheetNumberTemplate = null,
-        string? endOfWorksheetMarkerTemplate = null,
-        string? rowPrefix = null,
-        string? columnSeparator = null,
-        string? rowSuffix = null,
-        string? blankCellValue = null)
-    {
-        this._withWorksheetNumber = withWorksheetNumber;
-        this._withEndOfWorksheetMarker = withEndOfWorksheetMarker;
-        this._withQuotes = withQuotes;
+    private readonly MsExcelConfig _config;
+    private readonly ILogger<MsExcelDecoder> _log;
 
-        this._worksheetNumberTemplate = worksheetNumberTemplate ?? DefaultSheetNumberTemplate;
-        this._endOfWorksheetMarkerTemplate = endOfWorksheetMarkerTemplate ?? DefaultEndOfSheetTemplate;
+    public IEnumerable<string> SupportedMimeTypes { get; } = new[] { MimeTypes.MsExcelX, MimeTypes.MsExcel };
```

Comment:
why add `MimeTypes.MsExcel` here if the format is not supported and the result is an exception?

Reply:
Because I have kept the original behavior that uses a specific log message for old Office 97-2003 file formats: https://github.com/microsoft/kernel-memory/blob/489e4c66f96b0f3f4320d140a46d69fa65a1a26a/service/Core/Handlers/TextExtractionHandler.cs#L173. If it is OK to use the standard message, then we can safely remove this.

Reply:
yes I think log message added by the handler is sufficient when it doesn't find a handler, in which case we don't even need the exception I guess